### PR TITLE
[DRAFT] Initialize project structure

### PR DESCRIPTION
**Pull request by mccolljr**
This PR initializes the project structure.

Originally I had intended to use `nx` to create a monorepo, but I spent 40 minutes just trying to get the initial applications created and install dependencies. There were conflicting peer dependency versions between `nx` itself, `nest`, and `vue`.

Since `nx` didn't work, I created two separate applications with their own `package.json` files.

I did some basic work in each application, as well:

- In the `backend` application, I created some basic scaffolding to connect web sockets together in board sessions, identified by `?board=xxx` in the query string when connecting

- In the `frontend` application, I created an svg canvas which responds to mouse input to reposition the viewport

Since these are two separate applications, I created a Taskfile.yml to help with doing tasks for both projects at once. (see https://taskfile.dev/ for information & to find out how to install the `task` binary. It is a very useful utility)

This can be the starting point from which we do additional work.

### CODE REVIEW

**Comment by mccolljr** [comment]
@girlnotbird this is how you tag someone - use the `@` symbol followed by their name. It will notify them that they've been tagged, and is a good way to make sure someone sees something important. In this case, I just made this PR ready for review.

File: frontend/src/App.vue [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+const windowWidth = ref(window.innerWidth);
+const windowHeight = ref(window.innerHeight);
+
+const viewBox = computed(() => {
+  return `${offsetX.value} ${offsetY.value} ${windowWidth.value} ${windowHeight.value}`;
+});
+
+const onWindowResized = () => {
+  windowWidth.value = window.innerWidth;
```
Comment (mccolljr) [inline frontend/src/App.vue:50]: You can add comments to the files in the file view.

This is just an example.

```suggestion
  windowWidth.value = window.innerWidth; // you can also suggest code changes by using the suggestion button
```

Comment (mccolljr) [reply frontend/src/App.vue:50]: Once a comment has been discussed to the satisfaction of everyone involved, the comment can be resolved.

File: frontend/src/App.vue [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  window.removeEventListener('mousemove', onMouseMove);
+});
+</script>
+
+<template>
+  <svg class="cursor" :style="cursorStyle" viewBox="0 0 32 32" width="32" height="32">
+    <path d="M0,0 L32,8 L16,16 L8,32 Z" fill="black"></path>
+  </svg>
+
+  <svg :width="windowWidth.value" :height="windowHeight.value" :viewBox="viewBox" class="board">
```
Comment (girlnotbird) [inline frontend/src/App.vue]:
```suggestion
  <svg :width="windowWidth" :height="windowHeight" :viewBox="viewBox" class="board">
  // removed .value on windowWidth and windowHeight, vue automatically unwraps the refs
```

**Comment by girlnotbird** [comment]
@mccolljr Pulled your branch and looked through everything. I like this Task tool we're using. Everything looks good, and your sample app works! One line (127) in App.vue broke the build so I suggested a change. After that, looks ready to merge!

**Review by girlnotbird — CHANGES_REQUESTED** [review]
Suggested a change in App.vue line 127:

```<svg :width="windowWidth.value" :height="windowHeight.value" :viewBox="viewBox" class="board">
  <svg :width="windowWidth" :height="windowHeight" :viewBox="viewBox" class="board">
  // removed .value on windowWidth and windowHeight, vue automatically unwraps the refs
```

File: frontend/src/App.vue (hunk 2 above)
Comment (mccolljr) [reply frontend/src/App.vue]: What on earth lol, I don't remember adding the `.value` here. I think it wasn't in the original commit so idk what happened.

Once I wrap up work for the day I'll fix and push, or if you're free I don't mind if you fix and push to this branch

**Review by girlnotbird — APPROVED** [review]

**Pull request merged**